src/VirtoCommerce.BackInStockModule.Data/BackgroundJobs/BackInStockNotificationJobService.cs

```diff
@@ -0,0 +1,158 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Hangfire;
+using Microsoft.Extensions.Logging;
+using VirtoCommerce.BackInStockModule.Core.BackgroundJobs;
+using VirtoCommerce.BackInStockModule.Core.Models;
+using VirtoCommerce.BackInStockModule.Core.Notifications;
+using VirtoCommerce.BackInStockModule.Core.Services;
+using VirtoCommerce.CatalogModule.Core.Model.Search;
+using VirtoCommerce.CatalogModule.Core.Search;
+using VirtoCommerce.CustomerModule.Core.Services;
+using VirtoCommerce.NotificationsModule.Core.Extensions;
+using VirtoCommerce.NotificationsModule.Core.Services;
+using VirtoCommerce.Platform.Core.Common;
+using VirtoCommerce.Platform.Core.Settings;
+using VirtoCommerce.StoreModule.Core.Model;
+using VirtoCommerce.StoreModule.Core.Services;
+using BackInStockSettings = VirtoCommerce.BackInStockModule.Core.ModuleConstants.Settings.General;
+
+namespace VirtoCommerce.BackInStockModule.Data.BackgroundJobs;
+
+public class BackInStockNotificationJobService(
+    ISettingsManager settingsManager,
+    INotificationSearchService notificationSearchService,
+    INotificationSender notificationSender,
+    IMemberResolver memberResolver,
+    IProductSearchService productSearchService,
+    IStoreService storeService,
+    IBackInStockSubscriptionSearchService backInStockSubscriptionSearchService,
+    ILogger<BackInStockNotificationJobService> logger)
+    : IBackInStockNotificationJobService
+{
+    public void EnqueueProductBackInStockNotifications(IList<string> inStockProductsIds)
+    {
+        try
+        {
+            BackgroundJob.Enqueue(() =>
+                EnqueueBatchOfEmailNotificationsForProductIds(inStockProductsIds));
+        }
+        catch (Exception ex)
+        {
+            logger.LogError(ex, "Cannot enqueue back in stock notifications background job");
+            throw;
+        }
+    }
+
+    public async Task EnqueueBatchOfEmailNotificationsForProductIds(IList<string> inStockProductsIds)
+    {
+        foreach (var productId in inStockProductsIds)
+        {
+            try
+            {
+                var subscriptions = await FetchAllBackInStockSubscriptionsForProduct(productId);
+                foreach (var subscription in subscriptions)
+                {
+                    BackgroundJob.Enqueue(() =>
```

Comment:
bad idea. it will not work normally on loaded instance. what will happen if you send 500 of 1000 and job fails? Hangfire doc says to use this way. Need to update in other modules.